src/main/java/com/google/sps/servlets/RegistrationServlet.java

```diff
@@ -146,6 +158,29 @@ public void createUserEntityAndPutInDatastore(DatastoreService ds, Entity entity
     ds.put(entity);
   }
 
+  /** Returns the blob key for the user-uploaded file */
+  private String getBlobKey(HttpServletRequest request, String name, BlobstoreService blobstoreService) {
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(name);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+        return null;
+    } 
+
+    // Form only contains a single file input, so get the first index.
+    String blobKey = blobKeys.get(0).getKeyString();
```

Comment:
Don't know if this really matters, but should we put this line after we check if the user submitted the form without a pfp?

Reply:
It doesn't matter, but it makes more sense to have it afterwards yeah